src/components/Toolbar.js

```diff
@@ -153,7 +153,15 @@ export const Toolbar = connect(({ contexts, cursor, toolbarOverlay, scrollPriori
                 onMouseOut={clearHoldTimer}
                 onTouchEnd={clearHoldTimer}
                 onTouchStart={() => startOverlayTimer(id)}
-                onClick={e => exec(e)}
+                onClick={e => {
+                  if (id === 'exportContext') {
+                    dispatch({ type: 'showModal', id: 'export' })
+                    dispatch({ type: 'exportExec', execFunc: shortcutById })
+                  }
+                  else {
+                    exec(e)
+                  }
+                }}
```

Comment:
Functionality that is specific to `exportContext` belongs in the `exec` function in [`/src/shortcuts/exportContext.js`](https://github.com/cybersemics/em/pull/329/files#diff-e296d30097ef51854d6518450618880a). This handler should remain generic and needs to simply call `exec(e)`.